spec/backend/adapters/base-record.spec.js

```diff
@@ -16,4 +16,44 @@ describe('Record', function () {
       })
     })
   })
+  describe('#constructor', function () {
+    beforeEach(function () {
+      this.params = {
+        auth: {
+          login: 'login',
+        },
+      }
+    })
+
+    it('returns empty object if params are not passed to the constructor', function () {
+      const record = new Record()
+      expect(record.params).to.deep.equal({})
+    })
+
+    it('stores flatten object params', function () {
+      const record = new Record(this.params)
+      const expectedResult = { 'auth.login': 'login' }
+      expect(record.params).to.deep.equal(expectedResult)
```

Comment:
I would write this in one line like that:

```javascript
expect(new Record({ auth: { login: 'login' } }).params).to.deep.equal({ 'auth.login': 'login' })
```

and you could remove entire before `beforeEach` along with lines 34 and 35 (10 lines in total less) - just saying :)